src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/RewriteMapParser.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite
+{
+    public static class RewriteMapParser
+    {
+        public static IISRewriteMapCollection Parse(XElement xmlRoot)
+        {
+            if (xmlRoot == null)
+            {
+                throw new ArgumentException(nameof(xmlRoot));
+            }
+
+            var mapsElement = xmlRoot.Descendants(RewriteTags.RewriteMaps).SingleOrDefault();
+            if (mapsElement == null)
+            {
+                return null;
```

Comment:
I don't think so. It's valid not to have a rewrite map. Am I missing your point?